### DISCUSSION TITLE
Trying to specify mux path for unix domains

### DISCUSSION THREAD

**Discussion by hdonnay** [wezterm/wezterm#5361] [Q&A]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Mutter (GNOME 46, Fedora Linux 40.20240429.0 (Silverblue))

### WezTerm version

20240203-110809-5046fc22 (Flatpak)

### Ask your question!

I'm trying to better integrate WezTerm and my (admittedly odd) workflow that's based on `toolbox`/`toolbx`.

What I would like to do is create a unix domain where the mux-server is spawned inside a `toolbox`. The issue I'm running into is that I can't see how to change the socket path for a domain while having a single config. Because `/run/user` and `$HOME` are shared across all the sandboxes, the various domains need separate sockets. I've tried:

- modifying the domain tables based on whether the current process is in the flatpak or not
- modifying the domain tables based on whether the current process has `wezterm.gui` or not
- spawning the `wezterm-mux-server` directly

These all result in the same socket path being used.

Am I being insufficiently clever with my config or is this not possible?

**Comment by wez** [comment]
it's really hard to say because there's not a lot of wezterm specific information included here, no error messages to interpret, and I'm not familiar with toolbox.

In theory, you should be able to use a `proxy_command` in the unix domain to bootstrap it, I'm guessing at something like this:

```lua
config.unix_domains = {
  {
     name = "something",
     proxy_command = {"toolbox", "run", "--container", "something", "wezterm", "cli", "proxy" },
   }
}
```

If `/run/user` is mapped from the host environment, then you can persuade wezterm to use an alternate location by changing the bootstrap to set `$XDG_RUNTIME_DIR` differently.  From wezterm's perspective it doesn't have to be under `/run/user`. You can pick any arbitrary path that makes sense in your environment.

```lua
     proxy_command = {"toolbox", "run", "--container", "something", "env", "XDG_RUNTIME_DIR=/something", "wezterm", "cli", "proxy" },
```

**Reply by hdonnay** [reply]
Ah, okay, I think fiddling with `XDG_RUNTIME_DIR` might fix what I'm seeing. I'm a little worried that's going to have knock-on effects in any spawned shell, but will have to try it out.

**Reply by wez** [reply]
Keep in mind that you can use https://wezfurlong.org/wezterm/config/lua/config/set_environment_variables.html to set XDG_RUNTIME_DIR differently in the shells that wezterm spawns

**Reply by hdonnay** [reply]
Ah, I've been trying to use the `serve` settings, not the `proxy` setting. Yeah, this seems to work the way I want with `proxy_command` instead of `serve_command`+`socket_path`.
